### PR TITLE
Fixed problems with backlight and guideline elements misalignment and moving row after the last position

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

There are some problems related to setting position of `backlight` and `guideline` elements when HOT is placed in some extra container which has been scrolled. This PR fixes the problem in such case. Some demo is provided [within the issue](https://github.com/handsontable/dev-handsontable/issues/501).

### How has this been tested?

Tested with some issues related to placing backlight and tried scrolling both window and extra containers.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/dev-handsontable/issues/501

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/src/plugins/manualRowMove/__tests__/manualRowMoveUI.spec.js [hunk 1]
```diff
@@ -120,6 +120,92 @@ describe('manualRowMove', () => {
...
+        spec().$wrapper = $(`
+          <div class="wrapper">
+            <div class="extra-div">
+              <div class="extra-content">SOME TEXT</div>
+            </div>
+          </div>
+        `);
+
+        spec().$container = $(`<div id="${id}"></div>`);
+        spec().$styleDomEleement = $('<style type="text/css"></style>');
```
Comment (budnix) [inline handsontable/src/plugins/manualRowMove/__tests__/manualRowMoveUI.spec.js:135]:
```suggestion
        spec().$styleDomElement = $('<style type="text/css"></style>');
```
Typo

**Review by budnix — CHANGES_REQUESTED** [review]
It works great 💪 Just a small typo to correct.

**Review by budnix — APPROVED** [review]

File: handsontable/src/plugins/manualRowMove/__tests__/manualRowMoveUI.spec.js [hunk 2]
```diff
@@ -120,6 +120,92 @@ describe('manualRowMove', () => {
...
+            overflow: auto;
+          }
+
+          .extra-content {
+            height: 500px;
+          }
+       `;
+
+        $(document.createTextNode(css)).appendTo(spec().$styleDomEleement);
+        $(spec().$styleDomEleement).appendTo('body');
```
Comment (budnix) [inline handsontable/src/plugins/manualRowMove/__tests__/manualRowMoveUI.spec.js:157]:
```suggestion
        $(document.createTextNode(css)).appendTo(spec().$styleDomElement);
        $(spec().$styleDomElement).appendTo('body');
```

**Review by budnix — CHANGES_REQUESTED** [review]

**Review by budnix — CHANGES_REQUESTED** [review]
The same typos to fix but in different lines

**Review by budnix — APPROVED** [review]

**Comment by AMBudnik** [comment]
Released in Handsontable v12.4.0

**Pull request merged**